### PR TITLE
Adding TikhonovSVD and TSVD example networks

### PR DESCRIPTION
**Pull request by jcollfont**
Added new example networks for TikhonovSVD and TSVD for completeness

### CODE REVIEW

**Comment by jessdtate** [comment]
I'll merge this after we merge the tikhonov code into scirun.

**Comment by jessdtate** [comment]
can you change the test_toolkit_file.py to reflect the new number of networks (line 68)? It's a hard coded check that we need to improve.  You will also need to rebuild the toolkit file using the bundle_toolkit code in the SCIRun/bin/SCIRun directory.

**Comment by jcollfont** [comment]
Updated the number of networks in test_toolkit_file.py. 

The bundle_toolkit includes the toolkit in the SCIRun UI, right? I hadn’t done this before.

You are right, all the new networks (including isotropy method) should be applied after the merge in SCIRun with the new Tikhonov modules

**Comment by jessdtate** [comment]
the bundle_toolkit script will make the toolkit to include in the SCIRun UI, yes.  It's an extra step, but it is at least scripted.  Just run 
`{path to scirun bin scirun}/bundle_toolkit Networks/FwdInvToolkit.toolkit  Networks`

**Comment by jcollfont** [comment]
got it! It worked :)

File: test_toolkit_file.py [hunk 1]
```diff
@@ -7,8 +7,8 @@ def verify_network_node(network_xml):
     #  when the network format changes, the version number will change.
     #  If one network is upgrade or created with the new version, the rest must be upgraded too.
     #  load and save each network to upgrade it.
-    #  Run the script resave_networks.py to upgrade all networks.  
-    assert len(network_xml) == 10 # update when version changes
+    #  Run the script resave_networks.py to upgrade all networks.
+    assert len(network_xml) == 15 # update when version changes
```
Comment (dcwhite) [inline test_toolkit_file.py]: This is not where you update number of networks. It's below...

File: test_toolkit_file.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_toolkit_file():
     # count is first:
     assert networks[0].tag == 'count'
     # update actual network count here:
-    assert networks[0].text == '11'
+    assert networks[0].text == '14'
```
Comment (dcwhite) [inline test_toolkit_file.py]: ...here. Please pay attention to the comments.

**Review by dcwhite — CHANGES_REQUESTED** [review]

File: Documentation/ECGTool_inverse.tex [hunk 3]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+        \hat{X}   &=& \sum_{k=1}^K \frac{\sigma_k}{\lambda^2 + \sigma_k^2} v_k u_k^T Y,
+    \label{eq:inverseSec_TikhonovSolutions1}
+    \end{eqnarray}
+    \end{center} 
+    where $Y$ are the ECG measurements, $X$ are the unknown potentials on the heart, $\lambda$ is the regularization parameter and $\sigma_k$, $u_k$ and $v_k$ are the singular values, left and right singular vectors of the forward matrix $A$ and $K$ is its rank.
+    
+    The \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php5/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTikhonovSVD}{{\tt SolveInverseProblemWithTikhonovSVD}} module is shown in \autoref{fig:tik_moduleSVD}.
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.7\textwidth]{ECGToolkitGuide_figures/TikhonovSVD_module.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:131]: was this file added to the repo?

File: Documentation/ECGTool_inverse.tex [hunk 4]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+    
+    The solutions to the TikhonovSVD regularization problem depend on the selection of the regularization parameter ($\lambda$).
+    There are multiple approaches in the literature that allow for its selection.
+    \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php5/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTikhonovSVD}{{\tt SolveInverseProblemWithTikhonovSVD}} is currently implemented so that the user can choose between a direct entry, selection using a slider and the L-curve method, described in \autoref{sec:math:regparam}.
+    These methods can be found by selecting the appropriate option in the drop-down menu within the ``Lambda Method'' panel as done in \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTikhonov}{{\tt SolveInverseProblemWithTikhonov}} (see \autoref{fig:tik_module_gui}).
+    
+    
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.9\textwidth]{ECGToolkitGuide_figures/TikhonovSVDNetwork.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:176]: This seems to be missing too

File: Documentation/ECGTool_inverse.tex [hunk 5]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+        \hat{X}   &=& \sum_{k=1}^Q \frac{1}{\sigma_k} v_k u_k^T Y,
+    \label{eq:inverseSec_TikhonovSolutions1}
+    \end{eqnarray}
+    \end{center} 
+    where $Y$ are the ECG measurements, $X$ are the unknown potentials on the heart, $\lambda$ is the regularization parameter and $\sigma_k$, $u_k$ and $v_k$ are the singular values, left and right singular vectors of the forward matrix $A$ and $Q$ is the truncation point.
+    
+    The module \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php5/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTSVD}{{\tt SolveInverseProblemWithTSVD}} is shown along its inputs and outputs in \autoref{fig:TSVD_module}
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.7\textwidth]{ECGToolkitGuide_figures/TSVD_module.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:200]: and this one

File: Documentation/ECGTool_inverse.tex [hunk 6]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+    
+    The solutions to the TSVD regularization problem depend on the selection of the regularization parameter ($Q$).
+    There are multiple approaches in the literature that allow for its selection.
+    \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php5/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTSVD}{{\tt SolveInverseProblemWithTSVD}} is currently implemented so that the user can choose between a direct entry, selection using a slider and the L-curve method, described in \autoref{sec:math:regparam}.
+    These methods can be found by selecting the appropriate option in the drop-down menu within the ``Lambda Method'' panel as done in \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTikhonov}{{\tt SolveInverseProblemWithTikhonov}} (see \autoref{fig:tik_module_gui}).
+
+
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.9\textwidth]{ECGToolkitGuide_figures/TSVDNetwork.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:246]: and this

File: Documentation/ECGTool_inverse.tex [hunk 7]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+    The GUI for this module, shown in \autoref{fig:IsotropyMethod_gui}, allows to select a submatrix from an input matrix. In the case of the isotropy method, the users are only interested in the ``end'' parameter from the column range selector (lower right entry) since it determined where to truncate the right singular vectors.
+    
+    This implementation of the Isotropy Method has extra parameters that determine the operation of the Tikhonov inverse. These parameters are specific to the \href{http://scirundocwiki.sci.utah.edu/SCIRunDocs/index.php/CIBC:Documentation:SCIRun:Reference:BioPSE:SolveInverseProblemWithTikhonov}{{\tt SolveInverseProblemWithTikhonov}} module and we refer the user to \autoref{sec:inverse:tikhonov} for more details.
+    
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.9\textwidth]{ECGToolkitGuide_figures/IsotropyMethod_networkExample.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:288]: this too

File: Documentation/ECGTool_inverse.tex [hunk 8]
```diff
@@ -5,381 +5,528 @@ \section{Overview}
...
+    
+    \noindent{\bf Outputs:}
+    \begin{enumerate}
+        \item Activation times ($\tau\in\Re^{M,1}$ )
+    \end{enumerate}
+    The only output of the method is the inverse solution in the form of an array of activation times.
+    
+    \begin{figure}
+        \begin{center}
+        \includegraphics[width=0.9\textwidth]{ECGToolkitGuide_figures/activationBasedNetwork.png}
```
Comment (jessdtate) [inline Documentation/ECGTool_inverse.tex:480]: this one

**Comment by jessdtate** [comment]
The images referenced are missing from the repo

**Pull request merged**